### ISSUE TITLE
Checkboxes appear as [?] characters on Windows

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [mrhooray/gulp-mocha-phantomjs#70]
On the Win10 cmd terminal the checkboxes for success appear as `[?]`-like characters.

With `phantomjs: { useColors: true }`:
![with colors](https://cloud.githubusercontent.com/assets/3335181/17082435/3a136270-5133-11e6-8f10-bcea5a5630d1.PNG)

Without colors:
![without colors](https://cloud.githubusercontent.com/assets/3335181/17082436/3a18d7be-5133-11e6-81a1-ac240d4b0fda.PNG)

Note that this is exclusive to `gulp-mocha-phantomjs`; `gulp-mocha` alone is fine:
![normal elsewhere](https://cloud.githubusercontent.com/assets/3335181/17082441/5074f0a6-5133-11e6-9f0e-e2fe703ba553.PNG)

**Comment by mrhooray** [comment]
@nathanboktae any insights on this? could it be the version of included mocha or pretty print lib uses a character not supported by the font?

**Comment by nathanboktae** [comment]
> could it be the version of included mocha or pretty print lib uses a character not supported by the font?

No it's the same `spec` reporter from mocha. I haven't used Windows since windows 7 and the terminal [didn't even support color](https://github.com/nathanboktae/mocha-phantomjs/pull/38) then so... I might try `mocha-phantomjs-core` directly to rule out node.js in the picture. I suspect it might be a phantomjs2 thing on windows.

**Comment by shellscape** [comment]
@mrhooray @nathanboktae might be able to leverage [log-symbols](https://github.com/sindresorhus/log-symbols) which accounts for the current OS

**Comment by nathanboktae** [comment]
No, that's a node.js package. Mocha runs in phantomjs, not node, in `mocha-phantomjs` and it's variants.

**Comment by shellscape** [comment]
The concept is the same. User Agent sniffing for OS make instead of the `os` module.

**Comment by nathanboktae** [comment]
That logic is specific to the `spec` runner in mocha. You have 3 options as I see it:

1. Submit a PR to Mocha with the logic in the spec reporter to have it do nasty user agent string parsing in a reporter that was only designed to run in node.
2. Fork out your own spec reporter like [`mocha-better-spec-reporter`](https://github.com/btd/mocha-better-spec-reporter) and add that logic.
3. Use a non-shitty terminal like [Cmder](http://cmder.net/)

Either way it's not the problem of the `mocha-phantomjs` family of packages.

**Comment by shellscape** [comment]
@nathanboktae I'm not sure if it was intended, but your reply sure comes across as unnecessarily aggressive.

I was adding comment to assist @JoshuaKGoldberg. I shall now :vanish:

**Comment by nathanboktae** [comment]
> your reply sure comes across as unnecessarily aggressive.

The "non-shitty terminal" comment was directed at Microsoft for having thousands of developers on windows for years and years, and never putting any real effort into improving the terminal while scratching their heads wondering why devs are all using macs these days. Even after Windows 10 and adding bash, it appears to still not be up to par. Meanwhile, Cmder is a one-dev show.

These issues are tricky because it's hard to understand what's at fault, and there can often be many solutions.

**Comment by JoshuaKGoldberg** [comment]
Thanks for the comment @shellscape :)

@nathanboktae do you plan on reporting the issue to Mocha? If you not, could you please provide at least a brief RCA here so I can? I'm not familiar with the underlying issues beyond "character encoding is hard."

**Comment by nathanboktae** [comment]
Good find of ariya/phantomjs#13134 ... perhaps it is a phantomjs 2 issue after all.

> do you plan on reporting the issue to Mocha?

no because it's not an issue in node, and running the spec reporter in the browser is probably not a supported scenario for them, and even then pushing tricky UA string parsing into it is probably not what they want to do. Especially given alternatives of 2 and 3 in my [list of options](https://github.com/mrhooray/gulp-mocha-phantomjs/issues/70#issuecomment-269833057) above.
